viper/parser/parser_utils.py

```diff
@@ -378,6 +379,18 @@ def add_variable_offset(parent, key):
         if isinstance(typ, ListType):
             subtype = typ.subtype
             sub = ['uclamplt', base_type_conversion(key, key.typ, BaseType('num')), typ.count]
+        elif isinstance(typ, MappingType) and isinstance(key.typ, ByteArrayType):
+            if not isinstance(typ.keytype, ByteArrayType) or (typ.keytype.maxlen < key.typ.maxlen):
+                raise TypeMismatchException('Mapping keys of bytes cannot be cast, use exact same bytes type of: %s', str(typ.keytype))
+            subtype = typ.valuetype
+            if len(key.args[0].args) == 3:  # handle bytes literal.
+                sub = LLLnode.from_list([
+                    'seq',
+                    key,
+                    ['sha3', ['add', key.args[0].args[-1], 32], ceil32(key.typ.maxlen)]
+                ])
```

Comment:
Eventually I'd like to see byte literals being handled off chain as much as possible, but that can be another PR.

Reply:
+1